src/vdaf/mastic.rs

```diff
@@ -379,32 +382,477 @@ where
+                &id,
+                &input_share.vidpf_key,
+                public_share,
+                prefix,
+                &mut cache_tree,
+                nonce,
+            )?;
+            xof.update(&value_share.proof);
+            output_shares.append(&mut value_share.share.0);
+        }
+        let root_share_opt = if agg_param.require_weight_check {
+            Some(self.vidpf.eval_root_with_cache(
+                &id,
+                &input_share.vidpf_key,
+                public_share,
+                &mut cache_tree,
+                nonce,
+            )?)
+        } else {
+            None
+        };
+
+        let szk_verify_opt = if let Some(root_share) = root_share_opt {
+            Some(self.szk.query(
+                root_share.as_ref(),
+                input_share.proof_share.clone(),
+                verify_key,
+                nonce,
+            )?)
+        } else {
+            None
+        };
+        let (prep_share, prep_state) =
+            if let Some((szk_query_share, szk_query_state)) = szk_verify_opt {
+                let verifier_len = szk_query_share.flp_verifier.len();
+                (
+                    MasticPrepareShare {
+                        vidpf_proof: xof.into_seed(),
+                        szk_query_share_opt: Some(szk_query_share),
+                    },
+                    MasticPrepareState {
+                        output_shares: MasticOutputShare::<T::Field>::from(output_shares),
+                        szk_query_state,
+                        verifier_len: Some(verifier_len),
+                    },
+                )
+            } else {
+                (
+                    MasticPrepareShare {
+                        vidpf_proof: xof.into_seed(),
+                        szk_query_share_opt: None,
+                    },
+                    MasticPrepareState {
+                        output_shares: MasticOutputShare::<T::Field>::from(output_shares),
+                        szk_query_state: None,
+                        verifier_len: None,
+                    },
+                )
+            };
+        Ok((prep_state, prep_share))
+    }
+
+    fn prepare_shares_to_prepare_message<
+        M: IntoIterator<Item = MasticPrepareShare<T::Field, SEED_SIZE>>,
+    >(
+        &self,
+        _agg_param: &MasticAggregationParam,
+        inputs: M,
+    ) -> Result<MasticPrepareMessage<T::Field, SEED_SIZE>, VdafError> {
+        let mut inputs_iter = inputs.into_iter();
+        let leader_share = inputs_iter.next().ok_or(VdafError::Uncategorized(
+            "No leader share received".to_string(),
+        ))?;
+        let helper_share = inputs_iter.next().ok_or(VdafError::Uncategorized(
+            "No helper share received".to_string(),
+        ))?;
+        if inputs_iter.next().is_some() {
+            return Err(VdafError::Uncategorized(
+                "more than 2 prepare shares".to_string(),
```

Comment:
nit: try to be consistent about punctuation in error messages. Here you begin with a lower case letter, elsewhere you start with an upper case letter.